lib/tests/CMakeLists.txt

```diff
@@ -29,14 +31,44 @@ endif()
 # described in https://gcc.gnu.org/ml/gcc-help/2015-08/msg00035.html, we get
 # the error undefined reference to symbol '__tls_get_addr@@GLIBC_2.3'.
 # Build mock_server as a separate shared library to avoid this error.
+
+# pcp-chunk provides cpp-pcp-clients functions needed by MockServer
+if (LEATHERMAN_SHARED)
+    # Careful with the order, as done for libcpp-pcp-client
+    list(APPEND PCP_CHUNK_LIBS
+        ${Boost_REGEX_LIBRARY}
+        ${Boost_SYSTEM_LIBRARY}
+        ${LEATHERMAN_LIBRARIES})
+else()
+    list(APPEND PCP_CHUNK_LIBS
+        ${LEATHERMAN_LIBRARIES}
+        ${Boost_REGEX_LIBRARY}
+        ${Boost_SYSTEM_LIBRARY})
+endif()
+
+add_library(pcp-chunks
+    ${SOURCE_DIR}/protocol/chunks.cc
+    ${SOURCE_DIR}/protocol/message.cc
+    ${SOURCE_DIR}/protocol/serialization.cc
+    ${SOURCE_DIR}/validator/schema.cc
+    ${SOURCE_DIR}/validator/validator.cc)
+target_link_libraries(pcp-chunks PRIVATE ${PCP_CHUNK_LIBS} ${PLATFORM_LIBS})
+add_dependencies(pcp-chunks valijson)
+
+# Add the libcpp-pcp-client export header
+target_compile_definitions(pcp-chunks PRIVATE -Dlibcpp_pcp_client_EXPORTS)
+
 add_library(mock-server unit/connector/mock_server.cc)
 target_link_libraries(mock-server PRIVATE
+    ${LEATHERMAN_LIBRARIES}
```

Comment:
Pretty sure we shouldn't add these with LEATHERMAN_SHARED on. We end up with multiple definitions on Windows with libleatherman_logging and libboost_thread. That can probably be fixed with the ordering games at line 36, but you also have to be careful of the issue noted below cpp-pcp-client-unittests.